x/nexus/keeper/general_message.go

```diff
@@ -34,9 +33,9 @@ func getProcessingMessageKey(destinationChain exported.ChainName, id string) key
 func (k Keeper) GenerateMessageID(ctx sdk.Context) (string, []byte, uint64) {
 	counter := utils.NewCounter[uint64](messageNonceKey, k.getStore(ctx))
 	nonce := counter.Incr(ctx)
-	hash := sha256.Sum256(ctx.TxBytes())
+	txHash := utils.GetTxHash(ctx)
 
-	return fmt.Sprintf("0x%s-%d", hex.EncodeToString(hash[:]), nonce), hash[:], nonce
+	return fmt.Sprintf("0x%s-%d", hex.EncodeToString(txHash), nonce), txHash, nonce
```

Comment:
```suggestion
	return GetTxHashAsHex(ctx), txHash, nonce
```
You literally added this in the same PR 😛

Reply:
I didn't change it since we need to return `txHash` as well, and calling the helper would cause it to hash the tx bytes again